src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md

```diff
@@ -18,6 +18,99 @@ Or directly in the C# project file:
+
+These components enable enriching and redacting HTTP Client request logs. They remove built-it HTTP Client logging.
+
+In order to use the redaction feature, you need to reference the `Microsoft.Extensions.Compliance.Redaction` package.
+
+The services can be registered using the following methods:
+
+```csharp
+public static IServiceCollection AddExtendedHttpClientLogging(this IServiceCollection services);
+public static IServiceCollection AddExtendedHttpClientLogging(this IServiceCollection services, IConfigurationSection section);
+public static IServiceCollection AddExtendedHttpClientLogging(this IServiceCollection services, Action<LoggingOptions> configure);
+public static IServiceCollection AddHttpClientLogEnricher<T>(this IServiceCollection services) where T : class, IHttpClientLogEnricher;
+```
+
+For example:
+
+```csharp
+var builder = Host.CreateApplicationBuilder(args);
+
+// Register IHttpClientFactory:
+builder.Services.AddHttpClient();
+
+// Register redaction services:
+builder.Services.AddRedaction();
+
+// Register HttpClient logging enrichment & redaction services:
+builder.Services.AddExtendedHttpClientLogging();
+
+var host = builder.Build();
+```
+
+You can also use the following extension methods to apply the logging to the specific `IHttpClientBuilder`:
+
+```csharp
+public static IHttpClientBuilder AddExtendedHttpClientLogging(this IHttpClientBuilder builder);
+public static IHttpClientBuilder AddExtendedHttpClientLogging(this IHttpClientBuilder builder, IConfigurationSection section);
+public static IHttpClientBuilder AddExtendedHttpClientLogging(this IHttpClientBuilder builder, Action<LoggingOptions> configure);
+```
+
+For example:
+
+```csharp
+var builder = Host.CreateApplicationBuilder(args);
+
+// Register redaction services:
+builder.Services.AddRedaction();
+
+// Register named HttpClient:
+var httpClientBuilder = builder.Services.AddHttpClient("MyNamedClient");
+
+// Configure named HttpClient to use logging enrichment & redaction:
+httpClientBuilder.AddExtendedHttpClientLogging();
+
+var host = builder.Build();
+```
+
+### Tracking HTTP Request Client Latency
+
+These components enable tracking and reporting the latency of HTTP Client request processing.
+
+The services can be registered using the following methods:
+
+```csharp
+public static IServiceCollection AddHttpClientLatencyTelemetry(this IServiceCollection services);
+public static IServiceCollection AddHttpClientLatencyTelemetry(this IServiceCollection services, IConfigurationSection section);
+public static IServiceCollection AddHttpClientLatencyTelemetry(this IServiceCollection services, Action<HttpClientLatencyTelemetryOptions> configure);
+```
+
+For example:
+
+```csharp
+var builder = Host.CreateApplicationBuilder(args);
+
+// Register IHttpClientFactory:
+builder.Services.AddHttpClient();
+
+// Register redaction services:
+builder.Services.AddRedaction();
+
+// Register latency context services:
+builder.Services.AddLatencyContext();
```

Comment:
This was also non-trivial as DI simply throws an exception if I forget to add this line. We definitely need to improve this.

Reply:
Tracked under #4675